pytorch_lightning/callbacks/progress/tqdm_progress.py

```diff
@@ -227,12 +244,12 @@ def init_predict_tqdm(self) -> Tqdm:
     def init_validation_tqdm(self) -> Tqdm:
         """Override this to customize the tqdm bar for validation."""
         # The main progress bar doesn't exist in `trainer.validate()`
-        has_main_bar = self._main_progress_bar is not None
+        has_main_bar = not self.trainer.state.fn == "validate"
         bar = Tqdm(
-            desc="Validating",
+            desc=self.validation_description,
             position=(2 * self.process_position + has_main_bar),
             disable=self.is_disabled,
-            leave=False,
+            leave=not has_main_bar,
```

Comment:
Why this change?

Reply:
with trainer.test/predict, we would need to display bar once it is finished, with `trainer.validate` this wasn't the case since this was the common hook for fit_val and just val. I have added tests for it.

Reply:
Good catch. Can you fix this in a separate PR with its own test?

Reply:
I know it sucks. It took me a while to get used to doing it too.

But it makes life much easier for reviewers, the person cherry-picking commits for a release and any community person auditing the commits

Reply:
good point.. but it's a pain since CI is blocked, and once it opens up.. incoming merge conflicts 😭